src/api/web3Provider.ts

```diff
@@ -3,9 +3,31 @@ import { windowLoaded, promisify } from 'utils'
 import { Account, Provider } from 'types'
 import Web3 from 'web3'
 
-const setupWeb3 = async (provider: Provider) => {
+export const setupWeb3 = async (provider?: Provider) => {
   await windowLoaded
 
+  if (!provider) {
+    let providerWeb3: any
+    if (window.ethereum) {
+      providerWeb3 = window.web3 = new Web3(window.ethereum)
```

Comment:
no need to `let providerWeb3`, do `const` below

Reply:
yep, legacy from shitty testing